src/txn/txn_rollback_to_stable.c

```diff
@@ -358,6 +360,84 @@ __txn_abort_newer_col_fix(
         __txn_abort_newer_insert(session, ins, rollback_timestamp);
 }
 
+/*
+ * __txn_abort_row_hs_updates_using_build_in_mem_page --
+ *     Abort updates on a history store using the in-memory build reconciled page of data store.
+ */
+static int
+__txn_abort_row_hs_updates_using_build_in_mem_page(WT_SESSION_IMPL *session, const void *image,
+  const uint8_t *addr, size_t addr_size, wt_timestamp_t rollback_timestamp)
+{
+    WT_DECL_RET;
+    WT_ITEM tmp;
+    WT_PAGE *mod_page;
+    WT_ROW *rip;
+    uint32_t i, page_flags;
+    const void *image_local;
+
+    /*
+     * Don't pass an allocated buffer to the underlying block read function, force allocation of new
+     * memory of the appropriate size.
+     */
+    WT_CLEAR(tmp);
+
+    image_local = image;
+
+    if (image_local == NULL) {
+        WT_RET(__wt_bt_read(session, &tmp, addr, addr_size));
+        image_local = tmp.data;
+    }
+
+    page_flags = WT_DATA_IN_ITEM(&tmp) ? WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED;
+    WT_ERR(__wt_page_inmem(session, NULL, image_local, page_flags, &mod_page));
```

Comment:
Is it important that this page is being read from disk by this function? i.e: would it be bad if we ran this on a page that was already in memory?

Reply:
No, this function doesn't read the disk image. It creates the page with the image that either present in the memory or read by the above code. There are no parallel threads that read this page into memory. So I don't see a problem.